### PR TITLE
Add storing medical information

### PR DESCRIPTION
**Pull request by sibinhho99-nus**

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    private final String ethnicity;
+
+    /**
+     * Constructs a {@code JsonAdaptedMedical} with the given prescription details.
+     */
+    @JsonCreator
+    public JsonAdaptedMedical(@JsonProperty("nric") String nric,
+                                @JsonProperty("age") String age,
+                                @JsonProperty("bloodType") String bloodType,
+                              @JsonProperty("medication") String medication,
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:46]: Perhaps these can be alligned?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+        if (nric == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Nric.class.getSimpleName()));
+        }
+        if (!Nric.isValidNric(nric)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Nric modelNric = new Nric(nric);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:103]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                            BloodType.class.getSimpleName())
+            );
+        }
+
+        if (!BloodType.isValidBloodType(bloodType)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final BloodType modelBloodType = new BloodType(bloodType);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:126]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                            Medication.class.getSimpleName())
+            );
+        }
+
+        if (!Medication.isValidMedication(medication)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Medication modelMedication = new Medication(medication);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:139]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        if (height == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Height.class.getSimpleName()));
+        }
+
+        if (!Height.isValidHeight(weight)) {
+            throw new IllegalValueException(Height.MESSAGE_CONSTRAINTS);
+        }
+        final Height modelHeight = new Height(height);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:159]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+                            Illnesses.class.getSimpleName())
+            );
+        }
+
+        if (!Illnesses.isValidIllnesses(illnesses)) {
+            throw new IllegalValueException(Illnesses.MESSAGE_CONSTRAINTS);
+        }
+        final Illnesses modelIllnesses = new Illnesses(illnesses);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:172]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                            ImmunizationHistory.class.getSimpleName())
+            );
+        }
+
+        if (!ImmunizationHistory.isValidImmunizationHistory(immunizationHistory)) {
+            throw new IllegalValueException(ImmunizationHistory.MESSAGE_CONSTRAINTS);
+        }
+        final ImmunizationHistory modelImmunizationHistory = new ImmunizationHistory(immunizationHistory);
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:210]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+                            Gender.class.getSimpleName())
+            );
+        }
+
+        if (!Gender.isValidGender(gender)) {
+            throw new IllegalValueException(Gender.MESSAGE_CONSTRAINTS);
+        }
+        final Gender modelGender = new Gender(gender);
+
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:224]: Extra line break?

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+                            Ethnicity.class.getSimpleName())
+            );
+        }
+
+        if (!Ethnicity.isValidEthnicity(ethnicity)) {
+            throw new IllegalValueException(Ethnicity.MESSAGE_CONSTRAINTS);
+        }
+
+        final Ethnicity modelEthnicity = new Ethnicity(ethnicity);
+
```
Comment (joey-chance) [inline src/main/java/seedu/address/storage/JsonAdaptedMedical.java:238]: Extra line break?

**Review by joey-chance — COMMENTED** [review]
Some minor cosmetic issues. LGTM!

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 9 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:238]: done

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 8 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:224]: done, removed

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 7 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:210]: done, removed

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 6 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:172]: done, removed

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 5 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:159]: done, removed

File: src/main/java/seedu/address/storage/JsonAdaptedMedical.java (hunk 1 above)
Comment (sibinhho99-nus) [reply src/main/java/seedu/address/storage/JsonAdaptedMedical.java:46]: done

**Review by joey-chance — APPROVED** [review]
LGTM! 👍

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 10]
```diff
@@ -97,6 +106,13 @@ public AddressBook toModelType() throws IllegalValueException {
             }
             addressBook.addContact(contact);
         }
+        for (JsonAdaptedMedical jsonAdaptedMedical: medicals) {
+            Medical medical = jsonAdaptedMedical.toModelType();
+            if (addressBook.hasMedical(medical)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_CONTACT);
```
Comment (sibinhho99-nus) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:112]: Should be MESSAGE_DUPLICATE_MEDICAL

Comment (sibinhho99-nus) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:112]: Should be `MESSAGE_DUPLICATE_MEDICAL`

**Pull request merged**